Rbk/83 mariadb

rbruckenstein opened the pull request:
I have made the requested changes and they seem to work.  Please test it out.

rbruckenstein committed:
Add default command to initialize a database when needed.

rbruckenstein committed:
added test for existance of a database and command to initialize one when needed.

rbruckenstein committed:
Removed un-needed variable from defaults/main.yml

rbruckenstein committed:
Renamed register variable to more meaningfull name.  Hard-coded initialization command and removed un-needed quotes.

rbruckenstein commented:
I checked and sockets seem to go into /tmp in a protected subdirectory.  For example  /tmp/ssh-UTYs2haZsS/ which is then owned by the user and mode 700.   I also see a similar setup in /var/run/user.

rbruckenstein committed:
cleaned up check for data_dir.  Also added a stop of the Database before trying to initialize.   This is needed to be sure the DB is  created correctly.

rbruckenstein committed:
Moved socket infomation to client-server section and made socket location a variable.

rbruckenstein committed:
added variable for location of the my.cnf configuration file.  Set default location to /etc.

rbruckenstein committed:
Modified tasks to only stop and restart server if the data directory has changed.

rbruckenstein committed:
removed un-needed check to see if service is running.

rbruckenstein committed:
Modified the test for a changed data_dir to strip off trailing slashes.

rbruckenstein committed:
fixed logic to handle changing the socket directory correctly.

rjbennett3 commented:
@luisico The latest changes in Robin's rbruckenstein:rbk/83-mariadb branch look good to me and meet our needs for being able to customize the datadir and socket. I think the changes should be safe to merge.

luisico committed:
Configuration section client-server is already present in /etc/my.cnf

Moving socket there.

luisico committed:
Don't make users have to specify the path twice

luisico committed:
File /etc/my.cnf is standard in mariadb

- Don't anticipate the users to change it, so removing the option.
- Add the config_file param to other tasks using ansible modules.

For reference, the workaround with config_file is needed due to a change in
mysql libs used by ansible (see https://github.com/ansible/ansible/issues/47736)

luisico committed:
Simplify logic for stopping mysqld service

Also prevent logging of the root password and make it run in check mode.

luisico committed:
Group tasks logically
